bundles/mapping/mapmodule/request/map.layer.handler.js

```diff
@@ -64,6 +67,41 @@ Oskari.clazz.define('map.layer.handler',
                 evt._creator = request._creator;
                 sandbox.notifyAll(evt);
             }
+        },
+        _loadLayerInfo: function (layer, addLayer) {
+            if (typeof layer.getDescribeLayerStatus !== 'function') {
+                // layer type doesn't support this
+                addLayer();
+                return;
+            }
+            const layerId = layer.getId();
+            if (isNaN(layerId)) {
+                // only layers that have numeric ids can have reasonable response for DescribeLayer
+                addLayer();
+                return;
+            }
+            const status = layer.getDescribeLayerStatus();
+            if (status === DESCRIBE_LAYER.LOADED || status === DESCRIBE_LAYER.PENDING) {
+                return;
```

Comment:
Shouldn't we call addLayer() when `status === LOADED` since there isn't any pending calls waiting for network call for DescribeLayer  to finish? And only return when `status === PENDING` since the handler that is waiting on the describe will call addLayer once the info has been loaded?

Reply:
fixed